gdbm fails with Intel 16.0.2

```
$ spack install -j16 --restage gdbm%intel@16.0.2
...
==> Created stage in /home/certik/repos/spack/var/spack/stage/gdbm-1.13-zyq65fjmx4zqrcderjlaly2j73lwtg47
==> No patches needed for gdbm
==> Building gdbm [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

5 errors found in build log:
     [ ... ]
     253         ^
     254   
     255   gdbmtool.c(1220): warning #144: a value of type "int (*)()" cannot be used to initialize an entity of type "int (*)(struct handler_param *, size_t={unsigned long} *)"
     256         checkdb, recover_handler, NULL,
     257         ^
     258   
  >> 259   input-rl.c(18): catastrophic error: cannot open source file "readline/readline.h"
     260     #include <readline/readline.h>
     261                                   ^
     262   
     263   compilation aborted for input-rl.c (code 4)
     264   Makefile:694: recipe for target 'input-rl.o' failed
  >> 265   make[3]: *** [input-rl.o] Error 4
     266   make[3]: *** Waiting for unfinished jobs....
     267   make[3]: Leaving directory '/home/certik/tmp/spack-stage/spack-stage-K2UKpU/gdbm-1.13/src'
     268   Makefile:483: recipe for target 'all' failed
  >> 269   make[2]: *** [all] Error 2
     270   make[2]: Leaving directory '/home/certik/tmp/spack-stage/spack-stage-K2UKpU/gdbm-1.13/src'
     271   Makefile:456: recipe for target 'all-recursive' failed
  >> 272   make[1]: *** [all-recursive] Error 1
     273   make[1]: Leaving directory '/home/certik/tmp/spack-stage/spack-stage-K2UKpU/gdbm-1.13'
     274   Makefile:387: recipe for target 'all' failed
  >> 275   make: *** [all] Error 2

See build log for details:
  /home/certik/repos/spack/var/spack/stage/gdbm-1.13-zyq65fjmx4zqrcderjlaly2j73lwtg47/gdbm-1.13/spack-build.out
```

--- a/var/spack/repos/builtin/packages/gdbm/package.py
+++ b/var/spack/repos/builtin/packages/gdbm/package.py
@@ -45,4 +45,9 @@ class Gdbm(AutotoolsPackage):
     depends_on("readline")
 
     def configure_args(self):
-        return ['--enable-libgdbm-compat']
+        config_args = [
+            '--enable-libgdbm-compat',
+            'CC=%s' % spack_cc
+        ]
+
+        return config_args

